### PR TITLE
Support IPV6 when establishing MQTT connection to the cloud

### PR DESCRIPTION
**Pull request by stelfrag**
##### Summary
- Replace the mqtt ipv4-only connect function with connect_to_this_ip46 
- Replace mw_malloc, mw_calloc, mw_free, mw_strdup, mw_realloc
- Remove ws_client_get_nonce, (generate and use a UUID instead)
- Replace statistics pthread_lock with spinlock

##### Test
- Start agent normally, it should connect to the cloud
- Test IPV6 connectivity (we have it enabled only in the **testing** environment)
- Check with direct connection
- Check via a proxy (add `proxy = http://host:port` in netdata.conf `[cloud]` section)
   - Ideally check with a proxy on a VM that supports IPV6 and one that doesn't
   - You can force IPV4 connection on an otherwise IPV6 capable system by editing /etc/gai.conf -- adjust 
     `precedence ::ffff:0:0/96`

### CODE REVIEW

File: src/aclk/mqtt_websockets/mqtt_wss_client.c [hunk 1]
```diff
@@ -545,74 +537,66 @@ int mqtt_wss_connect(mqtt_wss_client client, char *host, int port, struct mqtt_c
...
-    if (client->sockfd < 0) {
-        mws_error(client->log, "Couldn't create socket()");
-        return -1;
+
+    char port_str[16];
+    snprintf(port_str, sizeof(port_str) -1, "%d", client->port);
+    int fd = connect_to_this_ip46(IPPROTO_TCP, SOCK_STREAM, client->host, 0, port_str, NULL);
+    if (fd < 0) {
+        mws_error(client->log, "Could not connect to remote endpoint \"%s\", port %d.\n", client->host, port);
+        return -3;
```
Comment (thiagoftsm) [inline src/aclk/mqtt_websockets/mqtt_wss_client.c:584]: Considering we are modifying the code now, could we also change the magic numbers like `-3` to some `define`?

File: src/aclk/mqtt_websockets/ws_client.c [hunk 2]
```diff
@@ -158,48 +161,27 @@ int ws_client_want_write(ws_client *client)
...
         return 1;
     }
 
-    ws_client_get_nonce(client, nonce, WEBSOCKET_NONCE_SIZE);
+    uuid_generate_random(nonce);
     EVP_EncodeBlock((unsigned char *)nonce_b64, (const unsigned char *)nonce, WEBSOCKET_NONCE_SIZE);
-    snprintf(second, TEMP_BUF_SIZE, websocket_upgrage_hdr,
-        *client->host,
-        nonce_b64);
+    snprintf(second, TEMP_BUF_SIZE, websocket_upgrage_hdr, *client->host, nonce_b64);
```
Comment (thiagoftsm) [inline src/aclk/mqtt_websockets/ws_client.c:183]: Two questions here:
- Why are we using `snprintf` instead our `snprintfz`?
- Considering that the majority of our code is using for size `size - 1`, should we do the same here?

File: src/aclk/mqtt_websockets/ws_client.c [hunk 3]
```diff
@@ -263,7 +245,7 @@ int ws_client_start_handshake(ws_client *client)
     if (rbuf_bytes_available(client->buf_read) < x)                                                                    \
         return WS_CLIENT_NEED_MORE_BYTES;
 
-#define MAX_HTTP_LINE_LENGTH 1024*4
+#define MAX_HTTP_LINE_LENGTH (1024 * 4)
```
Comment (thiagoftsm) [inline src/aclk/mqtt_websockets/ws_client.c:248]: A suggestion: we could use directly 4096 instead to do a multiplication considering this is a well known value in different OS.

**Comment by thiagoftsm** [comment]
@stelfrag except for the questions above, everything looks OK after these changes.

File: src/aclk/mqtt_websockets/ws_client.c (hunk 2 above)
Comment (stelfrag) [reply src/aclk/mqtt_websockets/ws_client.c:183]: I have another PR that further improves this (eg. removing pthread_locks) so I will add these changes to that PR

File: src/aclk/mqtt_websockets/ws_client.c (hunk 3 above)
Comment (stelfrag) [reply src/aclk/mqtt_websockets/ws_client.c:248]: I will do in a future PR

File: src/aclk/mqtt_websockets/mqtt_wss_client.c (hunk 1 above)
Comment (stelfrag) [reply src/aclk/mqtt_websockets/mqtt_wss_client.c:584]: I will do in a future PR

**Review by thiagoftsm — APPROVED** [review]
After comments, LGTM!

**Pull request merged**